chainer/testing/parameterized.py

```diff
@@ -85,10 +85,32 @@ def f(klass):
 
 
 def product(parameter):
-    keys = sorted(parameter)
-    values = [parameter[key] for key in keys]
-    values_product = itertools.product(*values)
-    return [dict(zip(keys, vals)) for vals in values_product]
+    if isinstance(parameter, dict):
+        keys = sorted(parameter)
+        values = [parameter[key] for key in keys]
+        values_product = itertools.product(*values)
+        return [dict(zip(keys, vals)) for vals in values_product]
+
+    elif isinstance(parameter, list):
+        # list of lists of dicts
+        if not all(isinstance(_, list) for _ in parameter):
+            raise TypeError('parameter must be list of lists of dicts')
+        if not all(isinstance(_, dict) for l in parameter for _ in l):
+            raise TypeError('parameter must be list of lists of dicts')
+
+        product = list(itertools.product(*parameter))
+        lst = []
+        for dict_lst in product:
```

Comment:
How about `for dict_lst in = list(itertools.product(*parameter)):` instead, not instantiating the `product` variable, partially because it shadows the function with the same name?

Reply:
Fixed.